lint fails for newly generated angular remote application

### Current Behavior

Newly generated remote using `npx nx generate @nx/angular:remote --name=newremote --directory=apps/newremote --projectNameAndRootFormat=as-provided --setParserOptionsProject=true --no-interactive` will fail `npx nx lint newremote` since eslint configuration with parserOptions is not configured to handle the .ts files coming from the module-federation config.

```
/Users/username/Documents/workspace/apps/newremote/webpack.prod.config.ts
  0:0  error  Parsing error: ESLint was configured to run on `<tsconfigRootDir>/apps/newremote/webpack.prod.config.ts` using `parserOptions.project`: 
- /users/username/documents/workspace/apps/newremote/tsconfig.app.json
- /users/username/documents/workspace/apps/newremote/tsconfig.editor.json
However, none of those TSConfigs include this file. Either:
- Change ESLint's list of included files to not include this file
- Change one of those TSConfigs to include this file
- Create a new TSConfig that includes this file and include it in your parserOptions.project
See the typescript-eslint docs for more info: https://typescript-eslint.io/linting/troubleshooting#i-get-errors-telling-me-eslint-was-configured-to-run--however-that-tsconfig-does-not--none-of-those-tsconfigs-include-this-file

```

### Expected Behavior

Newly generated remote using `npx nx generate @nx/angular:remote --name=newremote --directory=apps/newremote --projectNameAndRootFormat=as-provided --setParserOptionsProject=true --no-interactive` should successfully run `npx nx lint newremote`.

### GitHub Repo

_No response_

### Steps to Reproduce

1. npx nx generate @nx/angular:remote --name=newremote --directory=apps/newremote --projectNameAndRootFormat=as-provided --setParserOptionsProject=true --no-interactive
2. npx nx lint newremote


### Nx Report

```shell
Node   : 18.15.0
   OS     : darwin-arm64
   npm    : 9.5.0
   
   nx                 : 17.0.2
   @nx/js             : 17.0.2
   @nx/jest           : 17.0.2
   @nx/linter         : 17.0.2
   @nx/eslint         : 17.0.2
   @nx/workspace      : 17.0.2
   @nx/angular        : 17.0.2
   @nx/cypress        : 17.0.2
   @nx/devkit         : 17.0.2
   @nx/eslint-plugin  : 17.0.2
   @nx/plugin         : 17.0.2
   @nrwl/tao          : 17.0.2
   @nx/vite           : 17.0.2
   @nx/web            : 17.0.2
   @nx/webpack        : 17.0.2
   typescript         : 5.1.6
   ---------------------------------------
```


### Failure Logs

```shell
/Users/username/Documents/workspace/apps/newremote/webpack.prod.config.ts
  0:0  error  Parsing error: ESLint was configured to run on `<tsconfigRootDir>/apps/newremote/webpack.prod.config.ts` using `parserOptions.project`: 
- /users/username/documents/workspace/apps/newremote/tsconfig.app.json
- /users/username/documents/workspace/apps/newremote/tsconfig.editor.json
However, none of those TSConfigs include this file. Either:
- Change ESLint's list of included files to not include this file
- Change one of those TSConfigs to include this file
- Create a new TSConfig that includes this file and include it in your parserOptions.project
See the typescript-eslint docs for more info: https://typescript-eslint.io/linting/troubleshooting#i-get-errors-telling-me-eslint-was-configured-to-run--however-that-tsconfig-does-not--none-of-those-tsconfigs-include-this-file
```


### Package Manager Version

_No response_

### Operating System

- [X] macOS
- [ ] Linux
- [ ] Windows
- [ ] Other (Please specify)

### Additional Information

I guess the fix should be to include the .ts files coming from module federation in the tsconfig. But I let you decide what is best.

--- a/docs/generated/packages/angular/generators/setup-mf.json
+++ b/docs/generated/packages/angular/generators/setup-mf.json
@@ -78,6 +78,11 @@
         "type": "boolean",
         "description": "Whether the module federation configuration and webpack configuration files should use TS.",
         "default": true
+      },
+      "setParserOptionsProject": {
+        "type": "boolean",
+        "description": "Whether or not to configure the ESLint `parserOptions.project` option. We do not do this by default for lint performance reasons.",
+        "default": false
       }
     },
     "required": ["appName", "mfType"],

--- a/packages/angular/src/generators/host/host.ts
+++ b/packages/angular/src/generators/host/host.ts
@@ -74,6 +74,7 @@ export async function hostInternal(tree: Tree, schema: Schema) {
     prefix: options.prefix,
     typescriptConfiguration,
     standalone: options.standalone,
+    setParserOptionsProject: options.setParserOptionsProject,
   });
 
   let installTasks = [appInstallTask];

--- a/packages/angular/src/generators/remote/remote.ts
+++ b/packages/angular/src/generators/remote/remote.ts
@@ -68,6 +68,7 @@ export async function remoteInternal(tree: Tree, schema: Schema) {
     standalone: options.standalone,
     prefix: options.prefix,
     typescriptConfiguration,
+    setParserOptionsProject: options.setParserOptionsProject,
   });
 
   const installSwcHelpersTask = addDependenciesToPackageJson(

--- a/packages/angular/src/generators/setup-mf/files/ts-webpack/tsconfig.lint.json__tmpl__
+++ b/packages/angular/src/generators/setup-mf/files/ts-webpack/tsconfig.lint.json__tmpl__
@@ -1,0 +1,12 @@
+{
+  "extends": "./tsconfig.json",
+  "compilerOptions": {
+    "target": "ES2020"
+  },
+  "include": [
+    "src/main.ts",
+    <% if(type === "remote") { %> "src/remote-entry/<% if(standalone) { %>entry.routes.ts", <% } else { %> entry.module.ts", <% } } %>
+    "webpack.config.ts",
+    "webpack.prod.config.ts"
+  ]
+}

--- a/packages/angular/src/generators/setup-mf/lib/generate-config.ts
+++ b/packages/angular/src/generators/setup-mf/lib/generate-config.ts
@@ -39,4 +39,8 @@ export function generateWebpackConfig(
       standalone: options.standalone,
     }
   );
+
+  if (!options.setParserOptionsProject) {
+    tree.delete(joinPathFragments(appRoot, 'tsconfig.lint.json'));
+  }
 }

--- a/packages/angular/src/generators/setup-mf/schema.d.ts
+++ b/packages/angular/src/generators/setup-mf/schema.d.ts
@@ -15,6 +15,7 @@ export interface Schema {
   standalone?: boolean;
   skipE2E?: boolean;
   typescriptConfiguration?: boolean;
+  setParserOptionsProject?: boolean;
 }
 
 export interface NormalizedOptions extends Schema {

--- a/packages/angular/src/generators/setup-mf/schema.json
+++ b/packages/angular/src/generators/setup-mf/schema.json
@@ -78,6 +78,11 @@
       "type": "boolean",
       "description": "Whether the module federation configuration and webpack configuration files should use TS.",
       "default": true
+    },
+    "setParserOptionsProject": {
+      "type": "boolean",
+      "description": "Whether or not to configure the ESLint `parserOptions.project` option. We do not do this by default for lint performance reasons.",
+      "default": false
     }
   },
   "required": ["appName", "mfType"],

--- a/packages/react/src/generators/host/files/common/tsconfig.lint.json__tmpl__
+++ b/packages/react/src/generators/host/files/common/tsconfig.lint.json__tmpl__
@@ -1,0 +1,19 @@
+{
+  "extends": "./tsconfig.json",
+  "compilerOptions": {
+    "outDir": "../../dist/out-tsc",
+    "types": [
+        "node",
+        "@nx/react/typings/cssmodule.d.ts",
+        "@nx/react/typings/image.d.ts"
+    ]
+  },
+  "include": [
+    "src/**/*.js",
+    "src/**/*.jsx",
+    "src/**/*.ts",
+    "src/**/*.tsx",
+    "webpack.config.ts",
+    "webpack.prod.config.ts"
+  ]
+}

--- a/packages/react/src/generators/host/host.ts
+++ b/packages/react/src/generators/host/host.ts
@@ -105,6 +105,12 @@ export async function hostGeneratorInternal(
     updateProjectConfiguration(host, options.projectName, projectConfig);
   }
 
+  if (!options.setParserOptionsProject) {
+    host.delete(
+      joinPathFragments(options.appProjectRoot, 'tsconfig.lint.json')
+    );
+  }
+
   if (!options.skipFormat) {
     await formatFiles(host);
   }

--- a/packages/react/src/generators/remote/files/module-federation-ssr-ts/tsconfig.lint.json__tmpl__
+++ b/packages/react/src/generators/remote/files/module-federation-ssr-ts/tsconfig.lint.json__tmpl__
@@ -1,0 +1,19 @@
+{
+  "extends": "./tsconfig.json",
+  "compilerOptions": {
+    "outDir": "../../dist/out-tsc",
+    "types": [
+        "node",
+        "@nx/react/typings/cssmodule.d.ts",
+        "@nx/react/typings/image.d.ts"
+    ]
+  },
+  "include": [
+    "src/**/*.js",
+    "src/**/*.jsx",
+    "src/**/*.ts",
+    "src/**/*.tsx",
+    "webpack.config.ts",
+    "webpack.prod.config.ts"
+  ]
+}

--- a/packages/react/src/generators/remote/files/module-federation-ts/tsconfig.lint.json__tmpl__
+++ b/packages/react/src/generators/remote/files/module-federation-ts/tsconfig.lint.json__tmpl__
@@ -1,0 +1,19 @@
+{
+  "extends": "./tsconfig.json",
+  "compilerOptions": {
+    "outDir": "../../dist/out-tsc",
+    "types": [
+        "node",
+        "@nx/react/typings/cssmodule.d.ts",
+        "@nx/react/typings/image.d.ts"
+    ]
+  },
+  "include": [
+    "src/**/*.js",
+    "src/**/*.jsx",
+    "src/**/*.ts",
+    "src/**/*.tsx",
+    "webpack.config.ts",
+    "webpack.prod.config.ts"
+  ]
+}

--- a/packages/react/src/generators/remote/remote.ts
+++ b/packages/react/src/generators/remote/remote.ts
@@ -119,6 +119,11 @@ export async function remoteGeneratorInternal(host: Tree, schema: Schema) {
     );
     updateProjectConfiguration(host, options.projectName, projectConfig);
   }
+  if (!options.setParserOptionsProject) {
+    host.delete(
+      joinPathFragments(options.appProjectRoot, 'tsconfig.lint.json')
+    );
+  }
 
   if (!options.skipFormat) {
     await formatFiles(host);

